### PR TITLE
chore: linter as a GitHub Action

### PR DESCRIPTION
**Pull request by xcoulon**
Run Go and YAML linters on GitHub Actions

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
also, I wonder if we need the `ci/prow/build` and `ci/prow/test` jobs at all: I don't think that the `go build github.com/codeready-toolchain/api/api/v1alpha1/` command builds anything at all since there is no `main.go` in that directory, and we don't have any test at all either. In anycase, if we really want to keep those jobs, I would suggest that we migrate them to GitHub Actions, along with the linter proposed in this PR. 

WDYT?

**Review by MatousJobanek — APPROVED** [review]
I'm fine with dropping build as well as test jobs. Previously, we had some tests there, but they were removed.

**Comment by xcoulon** [comment]
> I'm fine with dropping build as well as test jobs. Previously, we had some tests there, but they were removed.

great, thanks for your feedback! I will take care of removing them

File: .golangci.yml [hunk 1]
```diff
@@ -1,10 +1,24 @@
 service:
-  project-path: github.com/codeready-toolchain/api
+  project-path: github.com/codeready-toolchain/host-operator
```
Comment (rajivnathan) [inline .golangci.yml:2]: Was this a mistake?

Comment (xcoulon) [reply .golangci.yml:2]: ah damn'it, I missed that :/

**Pull request merged**